### PR TITLE
CLC-5191 Mark bCourses user accounts which no longer match a CalNet UID

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.ets.berkeley.edu/jira/browse/CLC-5191

### CODE REVIEW

**Comment by johncrossman** [comment]
There is one happy Travis. I'll review.

File: app/models/canvas_csv/maintain_users.rb [hunk 1]
```diff
@@ -92,21 +100,56 @@ def handle_changed_sis_user_ids
...
+        end
+      end
+    end
+
     def categorize_user_account(existing_account, campus_user_rows)
       # Convert from CSV::Row for easier manipulation.
       old_account_data = existing_account.to_hash
       login_id = old_account_data['login_id']
+      if (inactive_account = /^inactive-([0-9]+)$/.match login_id)
+        login_id = inactive_account[1]
```
Comment (johncrossman) [inline app/models/canvas_csv/maintain_users.rb]: Is inactive_account[1] correct or do we want [0]?

Comment (raydavis) [inline app/models/canvas_csv/maintain_users.rb]: Yeah, I always have to look that up... [0] means "the whole original string"; [1] means "the first matched group".

File: app/models/canvas_csv/maintain_users.rb [hunk 2]
```diff
@@ -60,6 +63,11 @@ def initialize(known_uids, sis_user_import_csv)
     def refresh_existing_user_accounts
       check_all_user_accounts
       handle_changed_sis_user_ids
+      if Settings.canvas_proxy.delete_bad_emails.present?
```
Comment (johncrossman) [inline app/models/canvas_csv/maintain_users.rb]: I should know this... If Settings.canvas_proxy.delete_bad_emails is 'false' in yml then Settings.canvas_proxy.delete_bad_emails.present? will equal false?

Comment (raydavis) [inline app/models/canvas_csv/maintain_users.rb]: Yep, ".blank?" means false, [], '', {}, or nil, and ".present?" means "not blank?".

**Comment by johncrossman** [comment]
This looks good to me, @raydavis  I'll merge once I get your answer on 'delete_bad_emails' config

**Pull request merged**